### PR TITLE
Cart and Checkout pages functionalities added

### PR DESCRIPTION
**Pull request by sazedul-haque**

### CODE REVIEW

**Review by sifat009 — APPROVED** [review]

File: ecommerce/CartController.php [hunk 1]
```diff
@@ -94,7 +94,7 @@ public function __construct( $register_hooks = true ) {
 			 *
 			 * @since 3.0.0
 			 */
-			add_action( 'wp_ajax_tutor_cart_delete', array( $this, 'tutor_cart_delete' ) );
+			add_action( 'wp_ajax_tutor_course_from_cart', array( $this, 'tutor_course_from_cart' ) );
```
Comment (shewa12) [inline ecommerce/CartController.php:97]: I think this`wp_ajax_tutor_course_from_cart` should be `wp_ajax_tutor_delete_course_from_cart `

File: ecommerce/CartController.php [hunk 2]
```diff
@@ -166,6 +166,40 @@ public function get_cart_by_id() {
...
+	}
+
+	/**
+	 * Get delete course from cart
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return void JSON response
+	 */
+	public function tutor_course_from_cart() {
```
Comment (shewa12) [inline ecommerce/CartController.php:187]: Same here method name to be clear

File: ecommerce/CartController.php [hunk 3]
```diff
@@ -166,6 +166,40 @@ public function get_cart_by_id() {
...
+		if ( ! tutor_utils()->is_nonce_verified() ) {
+			$this->json_response( tutor_utils()->error_message( 'nonce' ), null, HttpHelper::STATUS_BAD_REQUEST );
```
Comment (shewa12) [inline ecommerce/CartController.php:189]: `$this->json_response( tutor_utils()->error_message( 'nonce' ), null, HttpHelper::STATUS_BAD_REQUEST );` I think this response is for API not for typical ajax request. Just use:
`wp_send_json_error( tutor_utils()->error_message( 'nonce' ) );`

**Review by shewa12 — CHANGES_REQUESTED** [review]
Please check the request changes

File: ecommerce/CartController.php (hunk 2 above)
Comment (sazedul-haque) [reply ecommerce/CartController.php]: Updated.

File: ecommerce/CartController.php (hunk 1 above)
Comment (sazedul-haque) [reply ecommerce/CartController.php]: Updated.

File: ecommerce/CartController.php (hunk 3 above)
Comment (sazedul-haque) [reply ecommerce/CartController.php]: Updated.

**Pull request merged**